dmsite/venv/src/store/actions/auth.js

```diff
@@ -0,0 +1,98 @@
+import axios from 'axios';
+
+import * as actionTypes from './actionTypes';
+
+
+export const authStart = () => {
+  return {
+    type: actionTypes.AUTH_START,
+  };
+};
+
+export const authSuccess = (token) => {
+  return {
+    type: actionTypes.AUTH_SUCCESS,
+    token,
+  };
+};
+
+export const authFail = (error) => {
+  return {
+    type: actionTypes.AUTH_FAIL,
+    error,
+  };
+};
+
+export const logout = () => {
+  localStorage.removeItem('token');
+  localStorage.removeItem('expirationDate');
+  return {
+    type: actionTypes.AUTH_LOGOUT,
+  };
+};
+
+export const checkAuthTimeout = (expirationDate) => {
+  return (dispatch) => {
+    setTimeout(() => {
+      dispatch(logout);
+    }, expirationDate * 1000);
+  };
+};
+
+export const authLogin = (username, password) => {
+  return (dispatch) => {
+    dispatch(authStart());
+    axios.post('http://localhost:8000/rest-auth/login', {
```

Comment:
Might want to add a comment to rename this if we move the domain name away from 'localhost'

Reply:
Same for the one below

Reply:
Actually, looking at it, it might be better to just define a constant that is the URL, so that when that URL changes, we only need to change it once instead of going through the code entirely multiple times.